python/benchmark/benchmark/bench_logistic_regression.py

```diff
@@ -0,0 +1,172 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from pyspark.ml.feature import StandardScaler
+from pyspark.ml.functions import array_to_vector, vector_to_array
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import col, sum
+
+from .base import BenchmarkBase
+from .utils import inspect_default_params_from_func, with_benchmark
+
+
+class BenchmarkLogisticRegression(BenchmarkBase):
+    def _supported_class_params(self) -> Dict[str, Any]:
+        from pyspark.ml.classification import LogisticRegression
+
+        params = inspect_default_params_from_func(
+            LogisticRegression.__init__,
+            [
+                "featuresCol",
+                "labelCol",
+                "predictionCol",
+                "weightCol",
+                "elasticNetParam",
+                "threshold",
+                "thresholds",
+                "aggregationDepth",
+                "maxBlockSizeInMB",
+                "lowerBoundsOnCoefficients",
+                "upperBoundsOnCoefficients",
+                "lowerBoundsOnIntercepts",
+                "upperBoundsOnIntercepts",
+            ],
+        )
+        return params
+
+    def run_once(
+        self,
+        spark: SparkSession,
+        train_df: DataFrame,
+        features_col: Union[str, List[str]],
+        transform_df: Optional[DataFrame],
+        label_name: Optional[str],
+    ) -> Dict[str, Any]:
+        assert label_name is not None
+
+        params = self.class_params
+        print(f"Passing {params} to LogisticRegression")
+
+        if self.args.num_gpus > 0:
+            from spark_rapids_ml.classification import LogisticRegression
+
+            lr = LogisticRegression(num_workers=self.args.num_gpus, **params)
+            benchmark_string = "Spark Rapids ML LogisticRegression training"
+
+        else:
+            from pyspark.ml.classification import (
+                LogisticRegression as SparkLogisticRegression,
+            )
+
+            lr = SparkLogisticRegression(**params)  # type: ignore[assignment]
+            benchmark_string = "Spark ML LogisticRegression training"
+
+        lr.setFeaturesCol(features_col)
+        lr.setLabelCol(label_name)
+
+        model, fit_time = with_benchmark(benchmark_string, lambda: lr.fit(train_df))
+
+        # placeholder try block till hasSummary is supported in gpu model
+        try:
+            if model.hasSummary:
+                print(f"total iterations: {model.summary.totalIterations}")
+                print(f"objective history: {model.summary.objectiveHistory}")
+        except:
+            print("model does not have hasSummary attribute")
```

Comment:
Probably would be better to implement a `hasSummary()` method in the model that just returns `False` for now.

Reply:
Revised in a new PR: https://github.com/NVIDIA/spark-rapids-ml/pull/375